.github/workflows/lint.yml

```diff
@@ -153,3 +153,20 @@ jobs:
           #   - Waiting for https://github.com/mackwic/colored/pull/122 to be merged.
 
           args: --file Cargo.lock --deny warnings --ignore RUSTSEC-2021-0145
+
+  no_std:
+    name: Check `no_std` compatability
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the Source Code
+        uses: actions/checkout@v3
+
+      - name: Set up Rust
+        uses: ./.github/actions/setup-rust
+
+      - name: Check
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --no-default-features -F serde,events,ledger_nano,message_interface,participation,rocksdb,serde,storage,stronghold,tls,rand
```

Comment:
I think the features are a bit confusing tbh, it works because client and wallet are not here so most of these feature aren't actually used but they are definitely not no_std

Reply:
Prefer to just check types then?

Reply:
Yeah I think so, I don't even think we'll ever be able to make anything else no_std anyway

Reply:
Maybe pow, not even sure